codegen/src/lib.rs

```diff
@@ -15,6 +15,37 @@
 // along with subxt.  If not, see <http://www.gnu.org/licenses/>.
 
 //! Library to generate an API for a Substrate runtime from its metadata.
+//!
+//! ## Usage Example
+//! ```rust
+//! use codec::Decode;
+//! use frame_metadata::RuntimeMetadataPrefixed;
+//! use subxt_codegen::DerivesRegistry;
+//!
+//! // Runtime metadata obtained from a node.
+//! let metadata = <RuntimeMetadataPrefixed as Decode>::decode(encoded)?;
+//! // Module under which the API is generated.
+//! let item_mod = syn::parse_quote!(
+//!     pub mod api {}
+//! );
+//! // Default module derivatives.
+//! let mut derives = DerivesRegistry::default();
+//! // Generate the Runtime API.
+//! let generator = subxt_codegen::RuntimeGenerator::new(metadata);
+//! let runtime_api = generator.generate_runtime(item_mod, derives);
+//! println!("{}", runtime_api);
+//! ```
```

Comment:
Just to be consistent with other Rust docs, perhaps move this to the end of the docs and call the title `# Example`?